docs/reports/marin-32b-retro.md

```diff
@@ -0,0 +1,353 @@
+| Warmup                 | 1% of steps                                                   |
+| Decay                  | 40% of steps                                                  |
+| Weight decay           | 0.05                                                          |
+| Max grad norm          | initially 1.0, then 0.2 from 56.4K steps                      |
+| Clip update norm       | on (σ=2.0, rolling=128); added at 72233, briefly off ~74k–80k |
+| EMA beta               | 0.995                                                         |
+| z-loss                 | 1e-4                                                          |
+| Skip bad steps         | true (σ=2.0, rolling=128)                                     |
+
+Some of these were activated mid-run due to loss spikes. These include the tightened max grad norm,
+the "clip update norm" and the "skip bad steps" flag. We discuss these below when we discuss the loss spikes.
+
+
+
+
+### Data Mix & Batch Schedule
+
+For pretraining, we followed the same recipe as our [8B Phoenix phase](./marin-8b-retro.md#phase-4-phoenix-reheated):
+[Nemotron-CC](https://arxiv.org/abs/2412.02595), [StarCoder Data](https://huggingface.co/datasets/bigcode/starcoderdata),
+and [Proofpile 2](https://huggingface.co/datasets/EleutherAI/proof-pile-2).
+
+Pretraining mixture (normalized share):
+
+| Dataset                     | Percentage |
+|----------------------------|------------|
+| nemotron_cc/medium         | 30.69%     |
+| nemotron_cc/hq_synth       | 24.70%     |
+| nemotron_cc/medium_low     | 13.98%     |
+| nemotron_cc/hq_actual      | 8.30%      |
+| nemotron_cc/medium_high    | 7.49%      |
+| nemotron_cc/low_actual     | 6.37%      |
+| nemotron_cc/low_synth      | 5.70%      |
+| starcoderdata              | 2.27%      |
+| proofpile_2                | 0.50%      |
+
+## Batch schedule
+
+
+| Start step | Global batch size | Tokens per batch |
+|------------|-------------------|------------------|
+| 0          | 8192              | 32Mi             |
+| 18,500     | 7680              | 30Mi             |
+| 21,010     | 8192              | 32Mi             |
+
+The change in global batch size was driven by changing hardware: when we had the 4 v5p-512s we used 8192,
+shifting down to 7680 for divisibility by 3 when we had 3 v5p-512s, and finally changing back to 8192
+when we moved to the v4 slice.
+
+As with the 8b run, we used a sequence length of 4096 tokens.
+
+## Training Phases
+
+![Training phases. See text for details.](../images/PLACEHOLDER_TIMELINE.png)
+
+### Phase 1: Scaling up our existing recipe
+
+For ~70k steps training behaved as expected though we had some loss spikes that people on Twitter/X told us to be worried about (Yay for open development!). While some other folks told us they looked fine, we added a bunch of clipping on update norms which seemed to mostly handle these spikes!
+
+Specifically, we looked at:
```

Comment:
Add some language from David's tweet - about how the loss was going down great, it recovered from spikes, so it seemed like it could be fine?  But we decided to look into it anyway...